go-controller/pkg/ovn/egressip.go

```diff
@@ -2374,7 +2379,7 @@ func (e *egressIPController) addExternalGWPodSNAT(podNamespace, podName string,
 // external GW setup in those cases.
 func (e *egressIPController) addExternalGWPodSNATOps(ops []ovsdb.Operation, podNamespace, podName string, status egressipv1.EgressIPStatusItem) ([]ovsdb.Operation, error) {
 	if config.Gateway.DisableSNATMultipleGWs {
-		if pod, err := e.watchFactory.GetPod(podNamespace, podName); err == nil && pod.Spec.NodeName == status.Node {
+		if pod, err := e.watchFactory.GetPod(podNamespace, podName); err == nil && pod.Spec.NodeName == status.Node && !util.PodCompleted(pod) {
```

Comment:
wondering if it might be good to change this addExternalGWPodSNATOps to pass in the pod itself instead of podNamespace and podName as strings, then we can avoid having to lookup anything in the informer cache (the functions that call it already have the pod object)